panoramapublic/test/src/org/labkey/test/components/panoramapublic/PanoramaPublicSearchWebPart.java

```diff
@@ -0,0 +1,149 @@
+import org.labkey.test.util.DataRegionTable;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class PanoramaPublicSearchWebPart extends BodyWebPart<PanoramaPublicSearchWebPart.ElementCache>
+{
+    private static String title = "Panorama Public Search";
+
+    public PanoramaPublicSearchWebPart(WebDriver driver, String title)
+    {
+        super(driver, title);
+    }
+
+    @Override
+    protected PanoramaPublicSearchWebPart.ElementCache newElementCache()
+    {
+        return new PanoramaPublicSearchWebPart.ElementCache();
+    }
+
+    public String getTitle()
+    {
+        return elementCache().title.get();
+    }
+
+    public PanoramaPublicSearchWebPart setTitle(String title)
+    {
+        elementCache().title.set(title);
+        return this;
+    }
+
+    public String getAuthor()
+    {
+        return elementCache().author.get();
+    }
+
+    public PanoramaPublicSearchWebPart setAuthor(String value)
+    {
+        elementCache().author.set(value);
+        return this;
+    }
+
+    public String getOrganism()
+    {
+        return elementCache().organism.get();
+    }
+
+    public PanoramaPublicSearchWebPart setOrganism(String value)
+    {
+        elementCache().organism.set(value);
+        return this;
+    }
+
+    public String getInstrument()
+    {
+        return elementCache().instrument.get();
+    }
+
+    public PanoramaPublicSearchWebPart setInstrument(String value)
+    {
+        elementCache().instrument.set(value);
+        return this;
+    }
+
+    public String getProtein()
+    {
+        return elementCache().protein.get();
+    }
+
+    public PanoramaPublicSearchWebPart setProtein(String value)
+    {
+        elementCache().protein.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart setProteinExactMatch(boolean value)
+    {
+        elementCache().proteinExactMatch.set(value);
+        return this;
+    }
+
+    public String getPeptide()
+    {
+        return elementCache().peptide.get();
+    }
+
+    public PanoramaPublicSearchWebPart setPeptide(String value)
+    {
+        elementCache().peptide.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart setPeptideExactMatch(boolean value)
+    {
+        elementCache().peptideExactMatch.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoExperimentSearch()
+    {
+        elementCache().experimentSearchTab.click();
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoProteinSearch()
+    {
+        elementCache().proteinTab.click();
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoPeptideSearch()
+    {
+        elementCache().peptideTab.click();
+        return this;
+    }
+
+    public DataRegionTable search()
+    {
+        elementCache().search.click();
+        return new DataRegionTable.DataRegionFinder(getWrapper().getDriver()).waitFor(this);
+    }
```

Comment:
Looking at the failure screenshot. It seems like searching creates a replacement data region instead of updating the existing one. Try waiting for the previous one to go stale when searching.
```suggestion
    public DataRegionTable search()
    {
        Optional<DataRegionTable> oldResults = new DataRegionTable.DataRegionFinder(getWrapper().getDriver()).findOptional(this)
        elementCache().search.click();
        if (oldResults.isPresent())
        {
            getWrapper().shortWait().until(ExpectedConditions.stalenessOf(oldResults.get().getComponentElement()));
        }
        return new DataRegionTable.DataRegionFinder(getWrapper().getDriver()).waitFor(this);
    }
```